webrender/src/renderer.rs

```diff
@@ -154,15 +154,22 @@ impl<L: GpuStoreLayout> GpuDataTexture<L> {
         }
 
         let items_per_row = L::items_per_row::<T>();
+        let rows_per_item = L::rows_per_item::<T>();
 
         // Extend the data array to be a multiple of the row size.
         // This ensures memory safety when the array is passed to
         // OpenGL to upload to the GPU.
-        while data.len() % items_per_row != 0 {
-            data.push(T::default());
+        if items_per_row != 0 {
+            while data.len() % items_per_row != 0 {
+                data.push(T::default());
+            }
         }
 
-        let height = data.len() / items_per_row;
+        let height = if items_per_row != 0 {
+            data.len() / items_per_row
+        } else {
+            data.len() * rows_per_item
```

Comment:
Multiplication is correct. We want the height to be the number of items divided by the number of items per row (which in our case would be 0.5), _but_ the existing `items_per_row` is an int. I was debating between this style (adding another integer `rows_per_item`) or just turning `items_per_row` into a float. They're both ugly, since `items_per_row` is primarily used by design as an int in order to ensure that the texture is a full rectangle, so I'd need to cast it to an int there and it all just felt nasty. I'm open to going with that option though.